src/com/sap/piper/analytics/Telemetry.groovy

```diff
@@ -0,0 +1,107 @@
+package com.sap.piper.analytics
+
+import com.cloudbees.groovy.cps.NonCPS
+import org.jenkinsci.plugins.workflow.steps.MissingContextVariableException
+
+class Telemetry implements Serializable{
```

Comment:
There is also a @Singleton annotation

Reply:
What does the annotation provide?

Reply:
I cannot use the annotation as it define a `final` `instance` variable that I can't clear during testing.